crypto/ed25519/chainkd/chainkd.go

```diff
@@ -30,71 +29,94 @@ func NewXPrv(r io.Reader) (xprv XPrv, err error) {
 	if err != nil {
 		return xprv, err
 	}
-	hasher := sha512.New()
-	hasher.Write([]byte("Chain seed"))
-	hasher.Write(entropy[:])
-	hasher.Sum(xprv[:0])
-	modifyScalar(xprv[:32])
-	return xprv, nil
+	return RootXPrv(entropy[:]), nil
 }
 
-func (xprv XPrv) XPub() XPub {
+// RootXPrv takes a seed binary string and produces a new XPrv.
+func RootXPrv(seed []byte) (xprv XPrv) {
```

Comment:
The reason was to explicitly define that "Root" key is generated differently from just any "Xprv". I'd suggest having these APIs for maximum clarity:

`RootXPrv([]byte)` - canonical root xprv interface

`RandomXPrv(io.Reader)` - convenience API to generate random root xprv (nil means default random generator `rand.Reader`)

WDYT?